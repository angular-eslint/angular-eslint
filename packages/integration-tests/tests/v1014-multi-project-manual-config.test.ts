import * as execa from 'execa';
import path from 'path';

const FIXTURES_DIR = path.join(__dirname, '../fixtures/');

function normalizeOutput(value: string): string {
  return value.replace(
    new RegExp(`^${FIXTURES_DIR.replace(/\\/g, '\\\\')}(.*?)$`, 'gm'),
    (_, c1) => `__ROOT__/${c1.replace(/\\/g, '/')}`,
  );
}

function runLint(directory: string): string | undefined {
  try {
    const cwd = path.join(FIXTURES_DIR, directory);
    process.chdir(cwd);

    const { stdout: lintOutput } = execa.sync('npx', ['ng', 'lint'], {
      cwd,
    });

    return normalizeOutput(lintOutput);
  } catch (error) {
    return normalizeOutput(error.stdout || error);
  }
}

const integrationTests: [string][] = [['v1014-multi-project-manual-config']];

describe.each(integrationTests)('%s', (directory) => {
  it('it should produce the expected lint output', () => {
    const lintOutput = runLint(directory);
    expect(lintOutput).toMatchSnapshot();
  });
});
