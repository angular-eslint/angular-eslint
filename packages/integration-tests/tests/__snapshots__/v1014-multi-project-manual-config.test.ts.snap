// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v1014-multi-project-manual-config it should produce the expected lint output 1`] = `
"
Linting \\"v1014-multi-project-manual-config\\"..."
`;
